service/src/main/python/jcarbon/tensorflow/callbacks.py

```diff
@@ -185,3 +188,100 @@ def on_epoch_end(self, epoch, logs=None):
         self.client.stop(self.pid)
         self.client.dump(
             self.pid, f'{self.output_path}/report-{epoch}.pb', self.signals)
+
+class JCarbonNvmlCallback(JCarbonCallback):
```

Comment:
Why is this inheriting from JCarbonCallback? This can just be NvmlSampleCallback. You can also move this class to the fourth in the list.